samples/c/create_systemd_scope.c

```diff
@@ -0,0 +1,614 @@
+	cg = cgroup_new_cgroup(MED_CGNAME);
+	if (!cg) {
+		printf("Failed to allocate the cgroup struct.  Are we out of memory?\n");
+		ret = ECGFAIL;
+		goto error;
+	}
+
+	ctrl = cgroup_add_controller(cg, "cpu");
+	if (!ctrl) {
+		printf("Failed to add the cpu controller to the %s cgroup struct\n",
+		       MED_CGNAME);
+		ret = ECGFAIL;
+		goto error;
+	}
+
+	ret = cgroup_set_value_string(ctrl, "cpu.weight", "300");
+	if (ret) {
+		printf("Failed to set %s's cpu.weight: %d: %s\n", MED_CGNAME, ret,
+		       cgroup_strerror(ret));
+		goto error;
+	}
+
+	ctrl = cgroup_add_controller(cg, "memory");
+	if (!ctrl) {
+		printf("Failed to add the memory controller to the %s cgroup struct\n",
+		       MED_CGNAME);
+		ret = ECGFAIL;
+		goto error;
+	}
+
+	ret = cgroup_set_value_string(ctrl, "memory.high", "3G");
+	if (ret) {
+		printf("Failed to set %s's memory.high: %d: %s\n", MED_CGNAME, ret,
+		       cgroup_strerror(ret));
+		goto error;
+	}
+
+	ret = cgroup_create_cgroup(cg, 0);
+	if (ret) {
+		printf("Failed to write the %s cgroup to /sys/fs/cgroup: %d: %s\n",
+		       MED_CGNAME, ret, cgroup_strerror(ret));
+		goto error;
+	}
+
+error:
+	if (cg)
+		cgroup_free(&cg);
+
+	return ret;
+}
+
+static int create_low_priority_cgroup(const struct example_opts * const opts)
+{
+	struct cgroup_controller *ctrl;
+	struct cgroup *cg;
+	int ret = 0;
+
+	cg = cgroup_new_cgroup(LOW_CGNAME);
+	if (!cg) {
+		printf("Failed to allocate the cgroup struct.  Are we out of memory?\n");
+		ret = ECGFAIL;
+		goto error;
+	}
+
+	ctrl = cgroup_add_controller(cg, "cpu");
+	if (!ctrl) {
+		printf("Failed to add the cpu controller to the %s cgroup struct\n",
+		       LOW_CGNAME);
+		ret = ECGFAIL;
+		goto error;
+	}
+
+	ret = cgroup_set_value_string(ctrl, "cpu.weight", "100");
+	if (ret) {
+		printf("Failed to set %s's cpu.weight: %d: %s\n", LOW_CGNAME, ret,
+		       cgroup_strerror(ret));
+		goto error;
+	}
+
+	ctrl = cgroup_add_controller(cg, "memory");
+	if (!ctrl) {
+		printf("Failed to add the memory controller to the %s cgroup struct\n",
+		       LOW_CGNAME);
+		ret = ECGFAIL;
+		goto error;
+	}
+
+	ret = cgroup_set_value_string(ctrl, "memory.max", "2G");
+	if (ret) {
+		printf("Failed to set %s's memory.high: %d: %s\n", LOW_CGNAME, ret,
+		       cgroup_strerror(ret));
+		goto error;
+	}
+
+	ret = cgroup_create_cgroup(cg, 0);
+	if (ret) {
+		printf("Failed to write the %s cgroup to /sys/fs/cgroup: %d: %s\n",
+		       LOW_CGNAME, ret, cgroup_strerror(ret));
+		goto error;
+	}
+
+error:
+	if (cg)
+		cgroup_free(&cg);
+
+	return ret;
+}
+
+static int create_process(pid_t * const pid, const char * const cgname)
+{
+	int ret = 0;
+
+	*pid = fork();
+	if (*pid < 0) {
+		printf("Failed to create the process: %d\n", errno);
+		ret = ECGFAIL;
+		goto error;
+	} else if (*pid == 0) {
+		/*
+		 * This is the child process.  Move it to the requested cgroup
+		 */
+		ret = cgroup_change_cgroup_path(cgname, getpid(), NULL);
+		if (ret)
+			exit(0);
+
+		while (true)
+			sleep(1);
```

Comment:
Should the sleep wake-up seconds be increased to larger number, instead of waking up every second

Reply:
Yeah, that makes sense.  I'll bump it up to `10` seconds.